tests/e2e/e2e_test.go

```diff
@@ -180,6 +186,28 @@ var _ = Describe("odoe2e", func() {
 		// TODO: Check if the application with the same name can be created
 	})
 
+	Context("should list applications in other project", func() {
+		It("should create a new project", func() {
+			session := runCmd("odo project create " + projName + "-2")
+			Expect(session).To(ContainSubstring(projName + "-2"))
+		})
+
+		It("should get the project", func() {
+			getProj := runCmd("odo project get --short")
+			Expect(strings.TrimSpace(getProj)).To(Equal(projName + "-2"))
+		})
+
+		It("should be able to list applications in other project", func() {
+			appNames := runCmd("odo app list --project " + projName)
```

Comment:
While you are actually currently checking the apps with --project, I feel you are still in the project $projName.
I think it would be more apt to set some other project as active and then trigger this command
So, please move this after https://github.com/redhat-developer/odo/blob/ea6a0cab4c9c5ce27f388f9f994f7e244b671e42/tests/e2e/e2e_test.go#L205-L208

Reply:
I am in `projName-2` and executing the `list --project projName` When you create a project it switches to new project same like openshift works. I think we are good here.

Reply:
>I am in projName-2 and executing the list --project projName 

aah ok :) I missed that "-2" there.. Thanks :)

>When you create a project it switches to new project same like openshift works. I think we are good here.

Yes .. My eyes missed the presence of "-2" being concatenated in the create line :) 
Thanks for pointing